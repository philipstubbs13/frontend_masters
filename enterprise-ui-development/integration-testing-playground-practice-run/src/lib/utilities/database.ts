import Prisma, * as PrismaAll from '@prisma/client';

const PrismaClient = Prisma?.PrismaClient || PrismaAll?.PrismaClient;

export let prisma;

if (process.env.NODE_ENV === 'production') {
  prisma = new PrismaClient();
} else {
  if (!global.prisma) {
    global.prisma = new PrismaClient();
  }
  prisma = global.prisma;
}
