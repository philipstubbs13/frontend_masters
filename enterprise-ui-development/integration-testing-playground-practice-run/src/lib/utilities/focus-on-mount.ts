export const focusOnMount = (node: HTMLElement) => {
  node.focus();
};
