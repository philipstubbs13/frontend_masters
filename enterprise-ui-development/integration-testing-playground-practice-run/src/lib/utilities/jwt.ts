export const decodeToken = (token: string) => Buffer.from(token, 'base64').toString('utf-8');
export const encodeToken = (token: string) => Buffer.from(token).toString('base64');
