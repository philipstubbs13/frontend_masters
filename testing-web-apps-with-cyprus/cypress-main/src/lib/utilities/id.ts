let current = 0;

export const getId = () => ++current;
