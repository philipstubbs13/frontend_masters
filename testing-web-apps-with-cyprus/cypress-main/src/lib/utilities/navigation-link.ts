import { page } from '$app/stores';

export const navigationLink = (node: HTMLLinkElement) => {
  const unsubscribe = page.subscribe(({ path }) => {
    if (path.startsWith(node.getAttribute('href'))) {
      node.dispatchEvent(new CustomEvent('routeChange', { detail: true }));
    } else {
      node.dispatchEvent(new CustomEvent('routeChange', { detail: false }));
    }
  });

  return {
    destroy() {
      unsubscribe();
    },
  };
};
