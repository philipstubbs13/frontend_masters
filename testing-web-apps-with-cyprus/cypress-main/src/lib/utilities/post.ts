export const post = (endpoint: string, data?: Parameters<typeof JSON.stringify>[0]) => {
  return fetch(endpoint, {
    method: 'POST',
    credentials: 'include',
    body: JSON.stringify(data || {}),
    headers: {
      'Content-Type': 'application/json',
    },
  }).then((r) => r.json());
};
