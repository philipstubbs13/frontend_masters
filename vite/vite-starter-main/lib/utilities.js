/**
 * Returns a random integer between min (inclusive) and max (inclusive).
 *
 * @param {number} min - The minimum value of the range.
 * @param {number} max - The maximum value of the range.
 * @returns {number} A random integer between min and max.
 */
export const getRandomNumber = (min, max) => {
  return Math.floor(Math.random() * (max - min + 1)) + min;
};

/**
 * Returns a random item from an array.
 *
 * @template T
 * @param {T[]} array - The array to pick a random item from.
 * @returns {T} A random item from the array.
 */
export const getRandomItemFromArray = (array) => {
  return array[getRandomNumber(0, array.length - 1)];
};

/**
 * Returns a new array with the specified number of random items from the original array.
 *
 * @template T
 * @param {T[]} array - The array to pick random items from.
 * @param {number} count - The number of random items to pick.
 * @returns {T[]} A new array with the specified number of random items from the original array.
 */
export const getRandomItemsFromArray = (array, count) => {
  const shuffled = array.sort(() => 0.5 - Math.random());
  return shuffled.slice(0, count);
};

/**
 * Returns a new array with the unique items from the original array.
 *
 * @template T
 * @param {T[]} array - The array to remove duplicates from.
 * @returns {T[]} A new array with the unique items from the original array.
 */
export const removeDuplicatesFromArray = (array) => {
  return [...new Set(array)];
};

/**
 * Returns a new object with the specified keys and values.
 *
 * @template T
 * @param {string[]} keys - The keys of the new object.
 * @param {T[]} values - The values of the new object.
 * @returns {Record<string, T>} A new object with the specified keys and values.
 */
export const zipArraysToObject = (keys, values) => {
  return keys.reduce((obj, key, index) => {
    obj[key] = values[index];
    return obj;
  }, {});
};
