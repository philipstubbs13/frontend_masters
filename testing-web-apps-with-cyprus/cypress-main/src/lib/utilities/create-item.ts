import { getId } from './id';

export const createItem = (title: string, packed = false): Item => ({
  id: getId(),
  title,
  packed,
});
