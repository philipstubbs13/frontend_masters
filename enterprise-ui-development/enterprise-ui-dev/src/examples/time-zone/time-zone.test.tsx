import { test, expect, vi } from 'vitest';
import { render } from 'test/utilities';
import TimeZone from '.';

test('it should render successfully', () => {
  render(<TimeZone />);
});

test.fails('should match the snapshot', async () => {
  const { container } = render(<TimeZone />);
  expect(container).toMatchSnapshot();
});
