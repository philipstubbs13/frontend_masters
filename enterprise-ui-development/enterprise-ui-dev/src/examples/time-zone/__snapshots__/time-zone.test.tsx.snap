// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should match the snapshot 1`] = `
<div>
  <section
    class="flex flex-col gap-8 rounded-lg border-4 border-primary-50 p-8"
  >
    <h1>
      1679492355195
    </h1>
    <p>
      You've been staring at this page for 
      0
       second(s).
    </p>
  </section>
</div>
`;
